analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -74,7 +74,12 @@ public void Execute()
                 return;
             }
             var current = queue.Dequeue();
-            if (visited.Add(current) && current.AddVisit() <= MaxOperationVisits)
+            if (visited.Add(current)
+                && current.AddVisit() is var visitCount
+                && (visitCount <= MaxOperationVisits
+                || (visitCount == MaxOperationVisits + 1
+                    && current.Block.ConditionalSuccessor is not null
```

Comment:
What's this line for? If I delete it, `RoslynSymbolicExecutionTest` is green.
```suggestion
```

Reply:
It does not affect logic. It avoids more complex checks for non-conditional blocks.